sql-plugin/src/main/scala/com/nvidia/spark/rapids/SortUtils.scala

```diff
@@ -407,3 +408,56 @@ class GpuSorter(
     }
   }
 }
+
+case class GpuSortOrderMeta(
+   sortOrder: SortOrder,
+   override val conf: RapidsConf,
+   parentOpt: Option[RapidsMeta[_, _, _]],
+   rule: DataFromReplacementRule
+) extends BaseExprMeta[SortOrder](sortOrder, conf, parentOpt, rule) {
+  override def tagExprForGpu(): Unit = {
+    if (isStructType(sortOrder.dataType)) {
+      val nullOrdering = sortOrder.nullOrdering
+      val directionDefaultNullOrdering = sortOrder.direction.defaultNullOrdering
+      val direction = sortOrder.direction.sql
+      if (nullOrdering != directionDefaultNullOrdering) {
+        willNotWorkOnGpu(s"only default null ordering $directionDefaultNullOrdering " +
+          s"for direction $direction is supported for nested types; actual: ${nullOrdering}")
+      }
+    }
+    if (isArrayOfStructType(sortOrder.dataType)) {
+      willNotWorkOnGpu("STRUCT is not supported as a child type for ARRAY, " +
+        s"actual data type: ${sortOrder.dataType}")
+    }
+  }
+
+  // One of the few expressions that are not replaced with a GPU version
+  override def convertToGpu(): Expression =
+    sortOrder.withNewChildren(childExprs.map(_.convertToGpu()))
+
+  private[this] def isStructType(dataType: DataType) = dataType match {
+    case StructType(_) => true
+    case _ => false
+  }
+
+  private[this] def isArrayOfStructType(dataType: DataType) = dataType match {
+    case ArrayType(elementType, _) =>
+      elementType match {
+        case StructType(_) => true
+        case _ => false
+      }
+    case _ => false
+  }
+}
+
+object GpuSortOrder {
+  val pluginChecks: ExprChecks = ExprChecks.projectOnly(
+    pluginSupportedOrderableSig + TypeSig.ARRAY.nested(gpuCommonTypes)
+      .withPsNote(TypeEnum.ARRAY, "STRUCT is not supported as a child type for ARRAY"),
+    TypeSig.orderable,
+    Seq(ParamCheck(
+      "input",
+      pluginSupportedOrderableSig + TypeSig.ARRAY.nested(gpuCommonTypes)
+        .withPsNote(TypeEnum.ARRAY, "STRUCT is not supported as a child type for ARRAY"),
+      TypeSig.orderable)))
+}
```

Comment:
I'm lukewarm on this kind of change.  We created yet another class just to hold the result of an ExprChecks call that otherwise wouldn't need a new class to hold it.  If we scale this out to all the exprs and execs we support, that's a lot of new classes that need to be loaded during startup.

Reply:
This makes sense too, updated